app/src/pages/inside/dashboardItemPage/newWidgets/charts/launchesComparisonChart/launchesComparisonChart.jsx

```diff
@@ -0,0 +1,277 @@
+    defaultMessage: 'System issue',
+  },
+  statistics$defects$no_defect: {
+    id: 'FilterNameById.statistics$defects$no_defect',
+    defaultMessage: 'No defect',
+  },
+  statistics$defects$to_investigate: {
+    id: 'FilterNameById.statistics$defects$to_investigate',
+    defaultMessage: 'To investigate',
+  },
+  ofTestCases: {
+    id: 'Widgets.ofTestCases',
+    defaultMessage: 'of test cases',
+  },
+});
+
+@injectIntl
+@connect(
+  (state) => ({
+    project: activeProjectSelector(state),
+  }),
+  {
+    redirect,
+  },
+)
+export class LaunchesComparisonChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    redirect: PropTypes.func.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    height: PropTypes.number,
+    project: PropTypes.string.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  componentWillUnmount() {
+    this.node.removeEventListener('mousemove', this.getCoords);
+  }
+
+  onChartCreated = (chart, element) => {
+    this.chart = chart;
+    this.node = element;
+
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    const legendClassName = cx('legend');
+    this.node.addEventListener('mousemove', this.getCoords);
+
+    d3
+      .select(this.node)
+      .insert('div', '.chart')
+      .attr('class', legendClassName)
+      .insert('div', `.${legendClassName}`)
+      .attr('data-js-legend-wrapper', '')
+      .selectAll('span')
+      .data(this.itemNames)
+      .enter()
+      .append('span')
+      .attr('data-id', (id) => id)
+      .attr('class', cx('legend-item'))
+      .html((id) => {
+        const color = chartConfigs.colors[id.split('$')[2]];
+        const itemName = this.props.intl.formatMessage(messages[id.split('$total')[0]]);
+        return `
+          <span class="${cx('color-mark')}" style="background-color: ${color}"></span>
+          <span class="${cx('item-name')}">${itemName}</span>`;
+      })
+      .on('mouseover', (id) => {
+        this.chart.focus(id);
+      })
+      .on('mouseout', () => {
+        this.chart.revert();
+      });
+
+    d3.selectAll(document.querySelectorAll('.c3-chart-bar path')).each(function() {
+      const elem = d3.select(this);
+      if (elem.datum().value === 0) {
+        elem.style('stroke-width', '3px');
+      }
+    });
+  };
+
+  onChartClick = (d) => {
+    const { widget, project } = this.props;
+
+    this.props.redirect({
+      type: TEST_ITEM_PAGE,
+      payload: {
+        projectId: project,
+        filterId: 'all',
+        testItemIds: widget.content.result[d.index].id,
+      },
+    });
+  };
+
+  getPosition = (d, width, height) => {
+    const rect = this.node.getBoundingClientRect();
+    const left = this.x - rect.left - width / 2;
+    const top = this.y - rect.top - height;
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getData = () => {
```

Comment:
Method started with `get*` but don't' return anything. It definitely should called somehow different.